### PR TITLE
build: add codecoverage report with codecov.io

### PR DESCRIPTION
**Pull request by cexbrayat**
This is an attempt to have codecoverage reports for both the backend and the frontend.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "ng": "ng",
     "start": "ng serve --proxy-config proxy.conf.json",
     "build": "ng build --prod",
-    "test": "ng test --single-run --code-coverage --no-progress --reporters dots",
+    "test": "ng test --single-run --code-coverage --no-progress --reporters dots,coverage-istanbul",
```
Comment (jnizet) [inline frontend/package.json]: coverage-istanbul is applied automatically when the --code-coverage option is passed, isn't it?

File: .travis.yml [hunk 2]
```diff
@@ -38,3 +38,13 @@ cache:
   # see https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#Caching-with-yarn
     - "frontend/.gradle/yarn"
     - $HOME/.cache/yarn
+
+script:
+  - ./gradlew check
```
Comment (jnizet) [inline .travis.yml]: No need to execute two scripts. Just execute the two tasks at once: `./gradlew check jacocoTestReport`.

**Comment by cexbrayat** [comment]
It works \o/ https://codecov.io/gh/Ninja-Squad/globe42/pull/75/tree

File: frontend/package.json (hunk 1 above)
Comment (cexbrayat) [inline frontend/package.json]: you're maybe right, I thought it was overriding the karma config, but I'll try.

**Comment by jnizet** [comment]
Almost: when you click on a file in the report, you get "Branch not found". Making it impossible to know what is being missed :-(

**Comment by cexbrayat** [comment]
Maybe because I pushed force : it looks like it's there now https://codecov.io/gh/Ninja-Squad/globe42/tree/build%2Fadd-coverage-info/frontend/src

**Comment by jnizet** [comment]
OK. I don't know where you clicked to get that (all the links generated by codecov in this PR lead to nowhere). But I guess it will work better once we merge it to master.

**Comment by jnizet** [comment]
Is it possible to disable the email notifications though?

**Comment by cexbrayat** [comment]
I may have found something, see previous commit inspired by https://github.com/codecov/support/wiki/Codecov-Yaml

**Pull request merged**